map_gen/maps/danger_ores/modules/expanse.lua

```diff
@@ -0,0 +1,486 @@
+      offset = offset + 1
+    end
+
+    chest_data[entity.unit_number] = { entity = entity, left_top = left_top, price = price }
+  end
+
+  local function process_chest(entity, budget)
+    if entity.name ~= 'logistic-chest-requester' then
+      return
+    end
+    if not chest_data[entity.unit_number] then
+      init_chest(entity, budget)
+    end
+
+    local container = chest_data[entity.unit_number]
+    if not container or not container.entity or not container.entity.valid then
+      chest_data[entity.unit_number] = nil
+      return
+    end
+
+    local inventory = container.entity.get_inventory(defines.inventory.chest)
+
+    if not inventory.is_empty() then
+      local contents = inventory.get_contents()
+      if contents['coin'] then
+        local count_removed = inventory.remove({ name = 'coin', count = 1 })
+        if count_removed > 0 then
+          remove_old_renders(container)
+          init_chest(entity, get_remaining_budget(container))
+          container = chest_data[entity.unit_number]
+          game.print({ 'danger_ores.chest_reset', { 'danger_ores.gps', math.floor(entity.position.x), math.floor(entity.position.y), entity.surface.name } })
+        end
+      end
+      if contents['infinity-chest'] then
+        remove_old_renders(container)
+        container.price = {}
+      end
+    end
+
+    for key, item_stack in pairs(container.price) do
+      local name = item_stack.name
+      local count_removed = inventory.remove({ name = name, count = item_stack.count })
+      container.price[key].count = container.price[key].count - count_removed
+      if container.price[key].count <= 0 then
+        remove_one_render(container, key)
+        table.remove(container.price, key)
+      end
+    end
+
+    if #container.price == 0 then
+      chest_data[entity.unit_number] = nil
+      if not inventory.is_empty() then
+        for name, count in pairs(inventory.get_contents()) do
+          entity.surface.spill_item_stack(entity.position, { name = name, count = count }, true, nil, false)
+        end
+      end
+      inventory.clear()
+
+      -- compute new chunk to unlock
+      local new_area
+      local step = 3
+      local vectors = { { -step, 0 }, { step, 0 }, { 0, step }, { 0, -step } }
+      table.shuffle_table(vectors) --left, right, down, up
+
+      for _, v in pairs(vectors) do
+        local tile = surface.get_tile({ entity.position.x + v[1], entity.position.y + v[2] })
+
+        if tile.name == 'out-of-map' then
+          local p = tile.position
+          new_area = {
+            left_top = {
+              x = p.x - p.x % cell_size,
+              y = p.y - p.y % cell_size,
+            },
+            right_bottom = {
+              x = p.x - p.x % cell_size + cell_size - 1,
+              y = p.y - p.y % cell_size + cell_size - 1,
+            }
+          }
+        end
+      end
+
+      if math.random() <= chance_to_receive_token then
```

Comment:
```suggestion
      if math.random() < chance_to_receive_token then
```
The probability is of this happening is basically zero, but `math.random()` can return 0, so without this change a `chance_to_receive_token`  of 0 could still spawn a coin. `math.random()` also return less than 1, so if the chance was set to 1 `math.random() < chance_to_receive_token` would always be true which is what we would want.